src/core/search/search_parser_test.cc

```diff
@@ -100,6 +129,70 @@ TEST_F(SearchParserTest, Parse) {
   EXPECT_EQ(1, Parse(" @foo: "));
 }
 
+TEST_F(SearchParserTest, MatchTerm) {
```

Comment:
I see all the tests have works separated by spaces, but we don't actually consider that with the actual search we perform. What's the behavior we aim for here?

Reply:
No specific matching behavior 🙂 I just want to make sure my parser + AST works. I can add a simple regex like \W+(term)\W+ as a half baked solution for now

RedisSearch uses indexed full text search, so all those cases should also match once/if we add it.

Reply:
We will not support full text search any time soon. The MVP of what we currently aim for is described here:https://github.com/dragonflydb/dragonfly/issues/431